File: Power of Two.py
#https://www.naukri.com/code360/problems/power-of-two_893061

def isPowerOfTwo(n:int) -> bool:
    # Write your code here
    while n:
        if n==1:
            return 1
        if n%2!=0:
            return 0
        n=n//2
    return 1

File: Power of Two.cpp
#https://leetcode.com/problems/power-of-two/description/

class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n==1){
            return 1;
        }
        if (n%2==0){
            long int val=1;
            while (1){
                if (n<val){
                    return 0;
                }
                else if(n==val){
                    return 1;

                }
                else{
                    val=val<<1;
                }
            }
        }
        return 0;
    }
};
